force-app/main/default/classes/MonthlyIterable_TEST.cls

```diff
@@ -538,4 +538,135 @@ public with sharing class MonthlyIterable_TEST {
+            DateTime.newInstance(2021, 1, 6, 13, 0, 0),
+            DateTime.newInstance(2021, 2, 3, 13, 0, 0),
+            DateTime.newInstance(2021, 3, 3, 13, 0, 0),
+            DateTime.newInstance(2021, 4, 7, 13, 0, 0)
+        };
+
+        MonthlyIterable iterable = new MonthlyIterable(dateTimeStart.date(), rRule);
+
+        Test.startTest();
+        List<DateTime> actual = new List<DateTime>();
+        while (iterable.hasNext()) {
+            actual.add(DateTime.newInstance(iterable.next(), datetimeStart.time()));
+        }
+        Test.stopTest();
+
+        System.assertEquals(
+            expected,
+            actual,
+            'List of Dates returned from the iterator was not as expected'
+        );
+    }
+
+    @IsTest
+    private static void shouldIterateEveryFirstWednesdayOfTheMonthWithInterval2OverAGivenYearUntilGivenEndDate() {
+        DateTime dateTimeStart = DateTime.newInstance(2020, 12, 2, 13, 0, 0);
+        rRule.withRuleString('FREQ=MONTHLY;UNTIL=20210502T000000Z;INTERVAL=2;BYDAY=+1WE');
+        List<DateTime> expected = new List<DateTime>{
+            DateTime.newInstance(2020, 12, 2, 13, 0, 0),
+            DateTime.newInstance(2021, 2, 3, 13, 0, 0),
+            DateTime.newInstance(2021, 4, 7, 13, 0, 0)
+        };
+
+        MonthlyIterable iterable = new MonthlyIterable(dateTimeStart.date(), rRule);
+
+        Test.startTest();
+        List<DateTime> actual = new List<DateTime>();
+        while (iterable.hasNext()) {
+            actual.add(DateTime.newInstance(iterable.next(), datetimeStart.time()));
+        }
+        Test.stopTest();
+
+        System.assertEquals(
+            expected,
+            actual,
+            'List of Dates returned from the iterator was not as expected'
+        );
+    }
+
+    @IsTest
+    private static void shouldIterateEveryFifthMondayOfTheMonthOverAGivenYearUntilGivenEndDate() {
+        DateTime dateTimeStart = DateTime.newInstance(2020, 12, 2, 13, 0, 0);
+        rRule.withRuleString('FREQ=MONTHLY;UNTIL=20210502T000000Z;INTERVAL=1;BYDAY=+5MO');
+        List<DateTime> expected = new List<DateTime>{
+            DateTime.newInstance(2021, 3, 29, 13, 0, 0)
+        };
+
+        MonthlyIterable iterable = new MonthlyIterable(dateTimeStart.date(), rRule);
+
+        Test.startTest();
+        List<DateTime> actual = new List<DateTime>();
+        while (iterable.hasNext()) {
+            actual.add(DateTime.newInstance(iterable.next(), datetimeStart.time()));
+        }
+        Test.stopTest();
+
+        System.assertEquals(
+            expected,
+            actual,
+            'List of Dates returned from the iterator was not as expected'
+        );
+    }
+
+    @IsTest
+    private static void shouldIterateEveryFifthMondayOfTheMonthUntilCount() {
+        DateTime dateTimeStart = DateTime.newInstance(2020, 12, 2, 13, 0, 0);
+        rRule.withRuleString('FREQ=MONTHLY;UNTIL=20210502T000000Z;BYDAY=+5MO;COUNT=5');
```

Comment:
This isn't right -- if we want to test count, we need to remove the UNTIL and update the expected.

Reply:
Shouldn't this be testing the scenario where it picks whichever comes first. Like if Until is provided it stops at until even if count is provided

Reply:
I think it's good to decide how to handle this.

https://tools.ietf.org/html/rfc5545#section-3.3.10
The UNTIL or COUNT rule parts are OPTIONAL, but they MUST NOT occur in the same 'recur'.

Reply:
I can update the tests . Thanks @AndersonTarren !!

Reply:
There's a default that we are testing elsewhere -- I believe it's UNTIL.